src/linear_algebra/affine_space/combination.lean

```diff
@@ -359,6 +385,94 @@ rfl
   ({i} : finset ι).centroid k p = p i :=
 by simp [centroid_def, affine_combination_apply]
 
+/-- The centroid of two points, expressed directly as adding a vector
+to a point. -/
+lemma centroid_insert_singleton [invertible (2 : k)] (p : ι → P) (i₁ i₂ : ι) :
+  ({i₁, i₂} : finset ι).centroid k p = (2 ⁻¹ : k) • (p i₂ -ᵥ p i₁) +ᵥ p i₁ :=
+begin
+  by_cases h : i₁ = i₂,
+  { simp [h] },
+  { have hc : (card ({i₁, i₂} : finset ι) : k) ≠ 0,
+    { rw [card_insert_of_not_mem (not_mem_singleton.2 h), card_singleton],
+      norm_num,
+      exact nonzero_of_invertible _ },
+    rw [centroid_def,
+        affine_combination_eq_weighted_vsub_of_point_vadd_of_sum_eq_one _ _ _
+          (sum_centroid_weights_eq_one_of_cast_card_ne_zero _ hc) (p i₁)],
+    simp [h],
+    norm_num }
+end
+
+/-- The centroid of two points indexed by `fin 2`, expressed directly
+as adding a vector to the first point. -/
+lemma centroid_insert_singleton_fin [invertible (2 : k)] (p : fin 2 → P) :
+  univ.centroid k p = (2 ⁻¹ : k) • (p 1 -ᵥ p 0) +ᵥ p 0 :=
+begin
+  have hu : (finset.univ : finset (fin 2)) = {0, 1}, by dec_trivial,
+  rw hu,
+  convert centroid_insert_singleton k p 0 1
+end
+
+/-- A centroid, over the image of an embedding, equals a centroid with
+the same points and weights over the original `finset`. -/
+lemma centroid_map (e : ι₂ ↪ ι) (p : ι → P) : (s₂.map e).centroid k p = s₂.centroid k (p ∘ e) :=
+by simp [centroid_def, affine_combination_map, centroid_weights]
+
+omit V
+
+/-- `centroid_weights` gives the weights for the centroid as a
+constant function, which is suitable when summing over the points
+whose centroid is being taken.  This function gives the weights in a
+form suitable for summing over a larger set of points, as an indicator
+function that is zero outside the set whose centroid is being taken.
+In the case of a `fintype`, the sum may be over `univ`. -/
+def centroid_weights_indicator : ι → k := set.indicator ↑s (s.centroid_weights k)
+
+/-- The definition of `centroid_weights_indicator`. -/
+lemma centroid_weights_indicator_def :
+  s.centroid_weights_indicator k = set.indicator ↑s (s.centroid_weights k) :=
+rfl
+
+/-- The sum of the weights for the centroid indexed by a `fintype`. -/
+lemma sum_centroid_weights_indicator [fintype ι] :
+  ∑ i, s.centroid_weights_indicator k i = ∑ i in s, s.centroid_weights k i :=
+(set.sum_indicator_subset _ (subset_univ _)).symm
+
+/-- In the characteristic zero case, the weights in the centroid
+indexed by a `fintype` sum to 1 if the number of points is not
+zero. -/
+lemma sum_centroid_weights_indicator_eq_one_of_card_ne_zero [char_zero k] [fintype ι]
```

Comment:
This doesn't seem to use the `char_zero` assumption, does it? Maybe also elsewhere?

Reply:
This uses `char_zero` via `sum_centroid_weights_eq_one_of_card_ne_zero`, which uses it via `nat.cast_eq_zero`, which is used by `simp` in proving that lemma.

Reply:
Ooh, you are right... I erronously looked at `sum_centroid_weights_eq_one_of_cast_card_ne_zero`.